### DISCUSSION TITLE
Matek h743-slim fails to connect to configurator

### DISCUSSION THREAD

**Discussion by ryet9** [iNavFlight/inav#7696] [Q&A] [answered]
## Current Behavior
Will not connect to configurator, DFU mode connects seamlessly, 3.0.2 works as intended every time

## Steps to Reproduce

1. flash 4.0.0 with full chip erase 
2. connect USB  **many times**
3. watch it time outs **many times**
4. cry deeply **many times**

## Expected behavior
a connection to the configurator

## Additional context
the same FC with firmware 3.0.2 will connect every time

---

can't get to CLI

- Matek h743-slim
- INAV version string:  4.0.0

**Comment by ZePapou** [comment]
I have quite the same behavior with this H743 slim v1.5 (Inav V4.0)
- when **no** SDCARD inserted : everything fine
- when SDCARD inserted : incorrect boot (blue led not flashing) - no connection with configurator
It was working with 3.0.2
I did not find anything on the net
Any help will be appreciated :)

**Comment by DzikuVx** [comment]
Tested both cases with my H743-slim and everything works fine 

<img width="1214" alt="image" src="https://user-images.githubusercontent.com/966811/147634509-caea7946-1c52-4429-b394-61a7c9f5e6dc.png">

No problems with flashing, SD card, etc

@ryet9 please verify if you have correct STM32 VCP driver installed
@ZePapou check if SD card is correctly placed and it's working with other FCs

**Comment by ZePapou** [comment]
Thanks, Pavel, for your support :)

The SDCARD is working well on an other FC (Matek F722se)
![F722](https://user-images.githubusercontent.com/96784361/147656912-c67cc970-07c8-4e70-bc45-eb762ba1da72.png)

I re flashed the 3.0.2 : boot succesfull even card not recognized in this version
I re flashed the 4.0 with SDCARD inserted in DFU mode : OK 
Failed to connect to configurator with SDCARD inserted, without : OK

Lost in INAV space !
:)

**Comment by ryet9** [comment, marked as answer]
just confirming that with different SDcards it will boot, took a while to get the right one, but it does indeed depend on the SDcard inserted
logging works as expected -- all is well with the world once again :)
currently using Toshiba V10 (10)/UHS 1 32gb

**Reply by LinkinL0gs** [reply]
LMAO, Wishing I had seen this discussion thread a week ago, might've saved me some sanity 🤤 [Live and you learn i spose](https://github.com/iNavFlight/inav/issues/7722)

**Comment by robustini** [comment]
Here the situation is worse than what I read above.
Without the MIcroSD the boot is ok, with some MicroSD it locks and you cannot connect, with others you connect but does not see the MicroSD formatted in FAT32.
Tested with iNav V5 and V6, same issue.
Hope @MATEKSYS  can solve the problem.

![immagine](https://user-images.githubusercontent.com/3864712/180600710-bec7a8e2-f8c0-4a8a-870c-2f67ce51fb55.png)

**Reply by robustini** [reply]
Betaflight, on the other hand, always boots but does not see the MicroSD, I have tried several.
Ardupilot instead sees and uses it regularly.

**Comment by robustini** [comment]
On seven MicroSD I found one that finally works, but this thing is quite castrating.
And the absurd thing is that this 4 Gb formatted regularly in FAT32 once it prepares the store space it goes to occupy it all, so it is unusable, look at the image.
So I think the target needs to be fixed for this board.

![immagine](https://user-images.githubusercontent.com/3864712/180637294-91a98ce0-dad8-440a-bfa7-f4fd37629269.png)

**Comment by robustini** [comment]
Other tests with iNav 5.1.0, MicroSD formatted on the PC, inserted in the slot, powered the FC and has been like this for 20 minutes.
I believe MicroSD support for this target needs to be completely revised.

![image](https://user-images.githubusercontent.com/3864712/181880344-83ff0937-983b-4dd0-b6c1-41ef5132a64a.png)

**Comment by robustini** [comment]
I think I have found the problem, and the solution, but I await confirmation from you.
The MicroSD that blocked iNav at boot I formatted them in "low level" all with Sd Card Formatter and now they all seem to work.
In addition iNav on first boot does not crash trying to allocate the space it will use for the logs but prepares them in no time.
However, it would be interesting if in any case the allocable space for the blackbox logs could be set.
I see that iNav never allocates more than 4 Gb, it would be useful to be able to declare the usable space in the MicroSD equal to or less than 4 Gb, for example it uses 2 Gb in a 4 Gb MicroSD.
I have to look in the code where this constraint is, and make sure it allocates half.

![immagine](https://user-images.githubusercontent.com/3864712/182014012-77b50575-e82c-476c-8d8e-b7a7e3fde14c.png)
